visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -21,6 +24,14 @@ export class CodeMapArrowService
 		CodeMapMouseEventService.subscribeToBuildingUnhovered(this.$rootScope, this)
 		ThreeSceneService.subscribeToBuildingSelectedEvents(this.$rootScope, this)
 		ThreeSceneService.subscribeToBuildingDeselectedEvents(this.$rootScope, this)
+
+		this.debounceCalculation = debounce(hoveredBuilding => {
```

Comment:
This looks good. But shouldn't we add a test in the spec-file to check whether debounce was used/ the debouncing-effect occurred at all?
Or would you "refer this matter to Q/A"?

Reply:
Seems to work more fluently now (on my local machine). :)

Reply:
I am fine adding a check before the wait in the test to verify that the debounce is working as expected.

Reply:
For the test, please check (the small) commit [95870f4](https://github.com/MaibornWolff/codecharta/pull/2897/commits/95870f4ba9ea6b67e32a5d56c871eb373f895b30) .

Much of it happened while pair-programming with Cedrik, but I would appreciate it if you gave me your advice too.